keras/layers/layer.py

```diff
@@ -509,6 +509,16 @@ def add_weight(
         self._track_variable(variable)
         return variable
 
+    def remove_weight(self, variable):
```

Comment:
Having `_untrack_variable` makes a lot of sense, but do we actually need a public `remove_weight`? Why not just `_untrack_variable`?

Reply:
Done.

The api `remove_weight` is redundant and might confuse users.